src/Admin/src/Adapter/AuthenticationAdapter.php

```diff
@@ -129,7 +129,7 @@ public function authenticate(): Result
                     ));
                 }
 
-                if ($identityClass->$methodName() !== $option['value']) {
+                if ($identityClass->$methodName()->value !== $option['value']) {
```

Comment:
```suggestion
                if ($identityClass->$methodName() !== $option['value']) {
```
We cannot force `->value` on `$methodName` because this can be a getter on any property from the `identity_class`. We can still check status as long as we use the enum in the option value. See below example:

`config/autoload/authentication.global.php`:
```php
'options'             => [
    'status' => [
        'value'   => \Admin\Admin\Enum\AdminStatusEnum::Active,
        'message' => \Admin\App\Message::ADMIN_INACTIVE,
    ],
],
```